src/main/java/org/zalando/baigan/service/aws/S3FileLoader.java

```diff
@@ -0,0 +1,100 @@
+package org.zalando.baigan.service.aws;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.services.kms.AWSKMS;
+import com.amazonaws.services.kms.AWSKMSClient;
+import com.amazonaws.services.kms.model.DecryptRequest;
+import com.amazonaws.services.s3.AmazonS3;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.google.common.io.BaseEncoding;
+
+import javax.annotation.Nonnull;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+/* Provides transparent content decryption of encrypted configuration content using AWS KMS. All configuration values
+ * surrounded by {@value #KMS_START_TAG} and {@value #KMS_END_TAG} are decrypted automatically. The content must
+ * be Base64 encoded and the decrypted content is interpreted as an UTF-8 encoded string. */
+
+public class S3FileLoader {
+
+    private static final String KMS_START_TAG = "awskms{";
+    private static final String KMS_END_TAG = "}";
+
+    /*
+     * Unfortunately necessary
+     */
+    private static final short MAX_RETRIES_BEFORE_EXCEPTION = 50;
+
+    private final AmazonS3 s3Client;
+    private final AWSKMS kmsClient;
+    private final String bucketName;
+    private final String key;
+
+    public S3FileLoader(@Nonnull String bucketName, @Nonnull String key) {
+        s3Client = new AmazonS3Client();
+        kmsClient = new AWSKMSClient();
+        this.bucketName = bucketName;
+        this.key = key;
+    }
+
+    public String loadContent() {
+        final String configurationText = s3Client.getObjectAsString(bucketName, key);
+        return decryptIfNecessary(configurationText);
+    }
+
+    private String decryptIfNecessary(final String candidate) {
+        final Optional<byte[]> encryptedValue = getEncryptedValue(candidate);
+        if (encryptedValue.isPresent()) {
+            return new String(toByteArray(decryptValue(encryptedValue.get())), StandardCharsets.UTF_8);
+        }
+       return candidate;
+    }
+
+    private ByteBuffer decryptValue(final byte[] encryptedBytes) {
+        final DecryptRequest request = new DecryptRequest().withCiphertextBlob(ByteBuffer.wrap(encryptedBytes));
+        short tries = 0;
+        while(true) {
+            try {
+                return kmsClient.decrypt(request).getPlaintext(); // This seems to randomly throw exceptions
```

Comment:
I've changed the retry logic to retry on `DependencyTimeoutException` and `KMSInternalException` only which should be enough...